### ISSUE TITLE
Nullable fields declared as `int` values default to 0, causing validation errors when doing a POST to update the thermostat's HvacMode

### ISSUE THREAD

**Issue by spencerwi** [i8beef/Ecobee#13]
When attempting the following request to change the hvacMode of a thermostat:

```c#
public async Task<Response> SetHvacMode(string thermostatId, string newMode)
{
	// TODO: this doesn't work for some reason
	try {
		var request = new ThermostatUpdateRequest
		{
			Selection = new Selection
			{
				SelectionType = "registered",
				SelectionMatch = thermostatId 
			},
			Thermostat = new Thermostat 
			{
				Settings = new Settings
				{
					HvacMode = newMode
				}
			}
		};
		var response = await this.ecobeeClient.PostAsync<ThermostatUpdateRequest, Response>(request);
		return response;
	} catch (ApiAuthException ex) {
		this.clearTokenSync();
		throw ex;
	}
}
```

I get this error response back from the Ecobee API:
```
7: Validation error. [Rule int(5,60) failed on: 0 /thermostat/ventilatorMinOnTime][Rule int(5,30) failed on: 0 /thermostat/setting/stage1HeatingDifferentialTemp][Rule int(5,30) failed on: 0 /thermostat/setting/stage1CoolingDifferentialTemp]
```

In particular, it's complaining that 0 is an invalid value for the following fields on the Settings object:
* `ventilatorMinOnTime`
* `stage1HeatingDifferentialTemp`
* `stage1CoolingDifferentialTemp`

At a glance, this error seems like it comes from nowhere: after all, the [Ecobee API docs list those fields as not-required](https://www.ecobee.com/home/developer/api/documentation/v1/objects/Settings.shtml), and I'm not providing them (just like in [their example](https://www.ecobee.com/home/developer/api/examples/ex8.shtml)), so it should work.

Glancing at the source of this lib, it looks like in the current version the [`Settings`](https://github.com/i8beef/Ecobee/blob/b7f6a372ea77c5cd36ad60b2fe7bac066c333ef2/src/I8Beef.Ecobee/Protocol/Objects/Settings.cs) object declares these fields as `int`, which is [a value type in C#](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types), so it can't ever have the value `null` -- instead, it seems like it's defaulting to `0`, which is what's causing the validation errors to occur.

Glancing around, it looks like there are quite a few fields that are declared as `int` when they probably should be `int?`, to avoid defaulting to 0 when the user does not intend to.

**Comment by i8beef** [comment]
Yup, that does seem like something that should be corrected. I'll take a look a tit this weekend, thank you.

**Comment by spencerwi** [comment]
No problem! Thanks again for putting out this lib, and for the quick response times -- so far, you're 2 for 2 with me on responding to issues _within a day_, which is spectacular for an OSS project.

**Comment by i8beef** [comment]
I just pushed a new version that should fix this (3.0.0.76)

**Comment by spencerwi** [comment]
That does indeed fix the issue. Thanks again!
